libbeat/publisher/client.go

```diff
@@ -103,27 +103,57 @@ func (c *client) PublishEvent(event common.MapStr, opts ...ClientOption) bool {
 		return false
 	}
 
-	ctx, pipeline := c.getPipeline(opts)
+	var values *outputs.Values
+	meta, ctx, pipeline := c.getPipeline(opts)
+	if len(meta) != 0 {
+		if len(meta) != 1 {
+			logp.Debug("publish", "too many metadata, pick first")
```

Comment:
Well it can happen. e.g. if 2 prospectors are configured and only one has `fields.pipeline` and the other not. So it might fail on purpose.